docs/decisions/20240103-use-jwt-claims-for-agent-admin-auth.md

```diff
@@ -0,0 +1,155 @@
+# Use JWT claims for agent admin access control
+
+- Status: accepted
+- Deciders: Pat Losoponkul, Yurii Shynbuiev, David Poltorak, Shailesh Patil
+- Date: 2024-01-03
+- Tags: multitenancy, authorisation, authentication
+
+Technical Story: [Allow agent admin role to be authenticated by Keycloak JWT | https://input-output.atlassian.net/browse/ATL-6074]
+
+## Context and Problem Statement
+
+External IAM authentication and authorization are currently limited to tenants using the wallets.
+Administrators currently rely on a static API key configured in the agent.
+Enhancing the security of admin access requires extending support for admin authentication through an external IAM.
+This introduces a concept of role to the authentication and authorization process.
+
+The existing tenant authentication and authorization model relies on UMA (user-managed access)
+which defines resources, policies, and permissions.
+Tenants gain access to resources based on defined permissions, effectively controlling resource access.
+In addition to wallet usage, the agent also handles wallet management,
+a functionality utilized by both tenants and administrators.
+While administrators don't directly use the wallet, they oversee its management.
+Integrating an auth model to distinguish between admins and tenants presents a new challenge.
+
+- Where and how to define the role of admin and tenant?
+- What boundary the admin role should be scoped to?
+- How to support different deployment topologies? (e.g. multi-agent in a shared realm, but not share the admin users)
```

Comment:
Making the authorization model agnostic to the deployment topologies would be nice. 
Multiple agents in a shared realm is a single case that should not significantly impact the current decision, from my point of view.
In real-world scenarios, user Groups and Roles will be managed by the external IAM system, so managing roles in the Keycloak client configuration and external system might be problematic.
I would focus on the most straightforward solution that can survive the new requirements as long as possible and can easily be replaced with another implementation.